semgrep_output_v1.atd

```diff
@@ -165,11 +165,44 @@ type error_severity
   | Info <json name="info">
 ]
 
-(* TODO? PRO can actually have "flavors" (see Engine_type.ml) *)
+(* Used for a best-effort report to users about what findings they get with
+   the pro engine that they couldn't with the oss engine.
+
+   Intrafile_taint = requires intrafile taint
+   Interfile_taint = requires interfile taint
+   Other_pro_feature = requires some non-taint pro feature *)
+type pro_feature
+  <ocaml attr="deriving show">
+  <python decorator="dataclass(frozen=True)"> = {
+    intraproc_taint: bool;
+    interfile_taint: bool;
+    proprietary_language: bool;
+}
+
+(* Report the engine used to detect each finding. Additionally, if we are able to
+   infer that the finding could only be detected using the pro engine, report that
+   the pro engine is required and include basic information about which feature is
+   required.
+
+   OSS = ran with OSS
+   PRO = ran with PRO, but we didn't infer that OSS couldn't have found this finding
+   (* Semgrep 1.64.0 or later *)
```

Comment:
move this comment abouce since semgrep 1.64.0 just before the PRO_REQUORED consturctor in the code.